### ISSUE TITLE
Cider optimization?

### ISSUE THREAD

**Issue by fawazsammani** [poojahira/image-captioning-bottom-up-top-down#3]
Hi. Thanks for the code!
Have you implemented Cider optimization as the author did? Or is it just Cross Entropy?
Thanks!

**Comment by poojahira** [comment]
No I did not implement Cider optimization. I imagine that would yield a better score.

**Comment by fawazsammani** [comment]
Thanks for your answer. Maybe I'll pull a request and do it when I have time. One more thing.... in your `datasets.py`, you've made the dataset for training and validation only, cause your hardcoded the location of the train and val images only. So in `eval.py`, when you are creating the loader: `loader = torch.utils.data.DataLoader(CaptionDataset(data_folder, data_name, 'TEST'), batch_size=1, shuffle=True, num_workers=1, pin_memory=torch.cuda.is_available())`, you are loading the validation features and not the testing features. There is no other file in your code to load the test features. So are the scores you reported on the GitHub page on the testing or validation images?
Thanks!

**Comment by poojahira** [comment]
The validation set of the MSCOCO dataset is split into val and test sets according to the famous Karpathy splits. This is done because the ground truth labels of the actual test set of MSCOCO is not publicly available and you need to submit your predictions to their evaluation server for results. The score reported is on the Karpathy test split. Hope this explains it.

**Comment by fawazsammani** [comment]
Sorry, forgot about that. Thanks!
I'll close this issue
